test/index.type-test.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { Text } from 'react-native'
+import styled from '../src/index'
+
+const StyledText = styled(Text)(({ transparent }: { transparent?: boolean; color: string }) => ({ flex: 1, opacity: transparent ? 0.5 : 1 }))
```

Comment:
@sparten11740 will it work with `styled.Text` ?

Reply:
we don't currently have the types for that defined, do we?

Reply:
ah overlooked them, they are in a different file

Reply:
i was thinking it's here

https://github.com/ExodusMovement/shakl/blob/befcafb83a22656d188c8715ead0ddbf3be2ee24/src/rn.tsx#L27

Reply:
for usage https://github.com/ExodusMovement/shakl/blob/befcafb83a22656d188c8715ead0ddbf3be2ee24/src/rn.tsx#L62

Reply:
I am a little confused by the type, it specifies we have `View, Text, Image, Touchable, TextInput, ScrollView` but we only define `View, Text, Image, Touchable`. what am I missing?

Reply:
https://github.com/ExodusMovement/shakl/pull/21/commits/ffd746730d2aac525ad915459a194e609396d738 is how the extended styles could be dealt with

Reply:
we don't need the explicit types besides passing in the generics to the styled function